### DISCUSSION TITLE
How to use Custom Type in Schema and the Client?

### DISCUSSION THREAD

**Discussion by ShubhamVerma1811** [prisma/prisma#9800] [Q&A]
Hi, I am new to Prisma.
My issue is that a field is of a particular type, it's not a model, its a field. And I want to get access to those custom types in the client.

It found this issue https://github.com/prisma/prisma/issues/2505, But not sure if this would help me.

```prisma
model Person {
  name String
// Custom type, in DB it could be a JSON[] but in prisma client it should show the Education types
  education  Education[]
}


type Education {
  collegeName String
}

```
Although education is stored as JSON in the DB, I know it would have a specific type structure that I want to access from the primsa client
```prisma
prisma.person.findMany({where:collegeName: foo})
```

Can this be acheived any how?

**Comment by nenadfilipovic** [comment]
I have few JSON fields in db and when I fetch them I must assign them type, by default type would be `Prisma.JsonValue`. After fetching and checking if value is present I assign type like this:

```ts
const { settings } = { ...data?.user };
const validSettings = settings ? (settings as AppPreferences) : undefined;
```

**Reply by ShubhamVerma1811** [reply]
So handling types from TS is the only way right now?

**Reply by nenadfilipovic** [reply]
I have used prisma-nexus so I had to use their Json type. This is what Prisma JSON type accepts, I don't think you can give Prisma your own type for Json field.

```ts
export declare type JsonValue =
  | string
  | number
  | boolean
  | null
  | JsonObject
  | JsonArray
```

Also check here how they assing types when trying to filter Json field https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields#reading-a-json-field

**Comment by ryands17** [comment]
@nenadfilipovic @ShubhamVerma1811 👋 

Json types cannot be typed directly in the schema as of now. We have a request for the same here so it would be great if you could add a 👍 so that we know the priority for this: https://github.com/prisma/prisma/issues/3219

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
